### ISSUE TITLE
JSDoc comment causes freeze

### ISSUE THREAD

**Issue by eloquence** [atom/language-javascript#540]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Trying to change the text "a.x" to "[a.x=" inside this JSDoc comment causes Atom to freeze.

````javascript
/**
* @param {x} a.x - *(default: none)* a b c
 */
````

Here's a little GIF that illustrates the effect:

![GIF animation](https://lib.reviews/static/uploads/atomfreeze.gif)

### Steps to Reproduce

1. Create a file that contains the text above
2. Attempt to change a.x to [a.x=

**Expected behavior:** Text is changed

**Actual behavior:** Atom freezes

**Reproduces how often:** 100%

### Versions

````
Atom: 1.21.2
Electron: 1.6.15
Chrome: 56.0.2924.87
Node: 7.4.0
OS: Ubuntu 17.04
````
### Additional Information

In addition to the recommended steps, I tried this with a clean install without packages. There's no output on the developer console.

**Comment by 50Wliu** [comment]
I couldn't reproduce this on Atom 1.23.0-dev-b6407650e.  That most likely means that it's been fixed in Atom 1.22 (currently in beta) or Atom 1.23-dev.  Thanks for completely filling out the issue template though; much appreciated :smile:.
![jsdoc-ok](https://user-images.githubusercontent.com/2766036/32353445-cafe36a4-c025-11e7-86da-e57473872c38.png)

**Comment by eloquence** [comment]
You're right, it works for me in the beta as well. Thanks for checking.
